src/components/MultiHopTrade/components/TradeInput/components/ManualAddressEntry.tsx

```diff
@@ -0,0 +1,112 @@
+import { FormControl, FormLabel } from '@chakra-ui/react'
+import { ethChainId } from '@shapeshiftoss/caip'
+import type { Dispatch, FC, SetStateAction } from 'react'
+import { useEffect, useMemo } from 'react'
+import { useFormContext } from 'react-hook-form'
+import { useTranslate } from 'react-polyglot'
+import { AddressInput } from 'components/Modals/Send/AddressInput/AddressInput'
+import { SendFormFields } from 'components/Modals/Send/SendCommon'
+import { getChainAdapterManager } from 'context/PluginProvider/chainAdapterSingleton'
+import { useFeatureFlag } from 'hooks/useFeatureFlag/useFeatureFlag'
+import { useWallet } from 'hooks/useWallet/useWallet'
+import { walletSupportsChain } from 'hooks/useWalletSupportsChain/useWalletSupportsChain'
+import { parseAddressInputWithChainId } from 'lib/address/address'
+import { selectBuyAsset } from 'state/slices/swappersSlice/selectors'
+import { swappers } from 'state/slices/swappersSlice/swappersSlice'
+import { useAppDispatch, useAppSelector } from 'state/store'
+
+type ManualAddressEntryProps = {
+  setIsManualAddressEntryValidating: Dispatch<SetStateAction<boolean>>
```

Comment:
This will keep types happy while making the component more dumb and decoupled from the parent logic
```suggestion
  setIsManualAddressEntryValidating: (isManualAddressEntryValidating: boolean) => void
```

Reply:
Good call, addressed in https://github.com/shapeshift/web/pull/4906/commits/985ddd9bee33e2d5c62a888344c64dc37f4ffbf8.